service/cert-signer/cert_signer.go

```diff
@@ -101,6 +128,115 @@ func (cs *certSigner) Issue(config spec.IssueConfig) (spec.IssueResponse, error)
 	return newIssueResponse, nil
 }
 
-func (cs *certSigner) SignedPath(clusterID string) string {
-	return fmt.Sprintf("pki-%s/issue/role-%s", clusterID, clusterID)
+func (cs *certSigner) SignedPath(clusterID string, organizations string) string {
+	return fmt.Sprintf("pki-%s/issue/%s", clusterID, roleName(clusterID, organizations))
+}
+
+func (cs *certSigner) IsRoleCreated(clusterID string, organizations string) (bool, error) {
```

Comment:
The PKI service already implements this so this should actually be reused. It would probably be odd to have the PKI service as a dependency of the cert signer so we probably want to have a role service that provides this. The role service would then be a dependency of the PKI service and the cert signer. The role service can then provide primitives like `Create` and do the checks if it exists and so on.